stl/inc/atomic

```diff
@@ -475,14 +475,33 @@ bool __stdcall _Atomic_wait_compare_non_lock_free(
 
 #ifdef _WIN64
 inline bool __stdcall _Atomic_wait_compare_16_bytes(const void* _Storage, void* _Comparand, size_t, void*) noexcept {
+
+#ifdef _M_ARM64
+#ifndef __clang__
     const auto _Dest              = static_cast<long long*>(const_cast<void*>(_Storage));
     const auto _Cmp               = static_cast<const long long*>(_Comparand);
     alignas(16) long long _Tmp[2] = {_Cmp[0], _Cmp[1]};
-#ifdef _M_X64
-    return _STD_COMPARE_EXCHANGE_128(_Dest, _Tmp[1], _Tmp[0], _Tmp) != 0;
-#else // ^^^ _M_X64 / ARM64 vvv
+
     return _InterlockedCompareExchange128_nf(_Dest, _Tmp[1], _Tmp[0], _Tmp) != 0;
-#endif // ^^^ ARM64 ^^^
+#else // ^^^ !__clang__ / __clang__ vvv
+    __int128* _Dest           = static_cast<__int128*>(const_cast<void*>(_Storage));
+    alignas(16) __int128 _Tmp = static_cast<const __int128*>(_Comparand)[0];

```

Comment:
Alignment of `_Tmp` shouldn't matter since `__builtin_arm_strex` takes its first argument by value. The alignment of `*_Storage` is what we should be concerned about - is it always aligned to at least `alignof(__int128)`?

Also, `meow[0]` is a peculiar way to spell `*meow` for a pointer `meow` with no arrays in sight.

`_Tmp` is never modified and so could be `const`.

Reply:
This code is only used with the `_Atomic_storage<_Ty&, 16>` class which aligns the storage to `alignas(16)`. The `__int128` type is also has 16 byte alignment so we're good there.

Reply:
Those facts match my recollection, thanks for verifying ;)